_posts/2021-05-09-april-rubygems-update.md

```diff
@@ -0,0 +1,43 @@
+---
+title: April 2021 RubyGems Updates
+layout: post
+author: Gift Egwuenu
+author_email: laurandidi21@gmail.com
+---
+
+Welcome to the RubyGems monthly update! As part of our efforts at [Ruby Together](http://rubytogether.org/), we publish a recap of the work that we’ve done the previous month. Read on to find out what updates were made to [RubyGems](https://github.com/rubygems/rubygems) and [RubyGems.org](https://github.com/rubygems/rubygems.org) in April.
+
+## RubyGems News
+This Month in RubyGems, we released a new version for **RubyGems(3.2.16)** and **Bundler(2.2.16)**. 
+
+RubyGems saw several bug fixes and updates this month, some of which include the following:
+
+- made sure specs are fetched from the right source when materializing - [#4562](https://github.com/rubygems/rubygems/pull/4562). 
+- merged a PR [#2836](https://github.com/rubygems/rubygems/pull/2836) that correctly handles symlinks when building gem packages by not adding any special handling.
+- improved `bundle pristine` error if `BUNDLE_GEMFILE` does not exist - [#4536](https://github.com/rubygems/rubygems/pull/4536).
+- fixed CI pipeline that was broken by third party releases of development dependencies.
+- fixed ``"Permission denied"`` error when running Bundler specs that resulted in finding `ruby-install + chruby` to be the fix to the error - [#3957](https://github.com/rubygems/rubygems/issues/3957).
+- fixed `bundle cache` with an up-to-date lockfile and specs not already installed - [#4554](https://github.com/rubygems/rubygems/pull/4554) 
```

Comment:
In my opinion, the updates mentioned here should only be things relevant to our users, so they should appear in the respective changelogs (https://github.com/rubygems/rubygems/blob/bb93b974100e9ddff7043e648d762e8a412be04e/bundler/CHANGELOG.md#2217-may-5-2021, https://github.com/rubygems/rubygems/blob/bb93b974100e9ddff7043e648d762e8a412be04e/CHANGELOG.md#3217--2021-05-05). In particular, the entry about "fixed CI pipeline" and the entry about "fixed `Permission denied`" error shouldn't be included since they are just quality of life improvements for bundler developers, but not interesting for users.

Instead, I would mention the following as "highlights" for this month:

> * We fixed an issue affecting custom `sidekiq-pro` gem servers, which was preventing users from upgrading their `sidekiq-pro` version.
> * We made `bundler` more secure by preventing any credentials from being logged to the screen, thus potentially preventing users from unintentionally leaking them when pasting them to a Github issue or similar situations.
> * We fixed a few resolution & materialization issues in bundler, and also improved rubygems handling of repositories including symlinks.

And I would include links to all releases made for more details.

Reply:
Great! For `sidekiq-pro` issue, it's this link: https://github.com/rubygems/rubygems/pull/4563.

For issues of redacting credentials, these: https://github.com/rubygems/rubygems/pull/4564, https://github.com/rubygems/rubygems/pull/4566.

For resolution and materialization issues, these: https://github.com/rubygems/rubygems/pull/4556, https://github.com/rubygems/rubygems/pull/4562. and for the symlinks rubygems issue, this one: https://github.com/rubygems/rubygems/pull/2836.

You can also mention the changelogs for all versions for more details:

* Rubygems 3.2.16: https://github.com/rubygems/rubygems/blob/bb93b974100e9ddff7043e648d762e8a412be04e/CHANGELOG.md#3216--2021-04-08
* Bundler 2.2.16: https://github.com/rubygems/rubygems/blob/bb93b974100e9ddff7043e648d762e8a412be04e/bundler/CHANGELOG.md#2216-april-8-2021
* Rubygems 3.2.17: https://github.com/rubygems/rubygems/blob/bb93b974100e9ddff7043e648d762e8a412be04e/CHANGELOG.md#3217--2021-05-05
* Bundler 2.2.17: https://github.com/rubygems/rubygems/blob/bb93b974100e9ddff7043e648d762e8a412be04e/bundler/CHANGELOG.md#2217-may-5-2021

Reply:
Thank you!